File: backend/src/main/java/com/back/domain/book/book/dto/BookDetailDto.java
package com.back.domain.book.book.dto;

import com.back.domain.bookmarks.constant.ReadState;
import com.back.domain.review.review.dto.ReviewResponseDto;
import com.back.global.dto.PageResponseDto;
import lombok.Builder;
import lombok.Getter;

import java.time.LocalDateTime;
import java.util.List;

@Getter
@Builder
public class BookDetailDto {
    private int id;
    private String title;
    private String imageUrl;
    private String publisher;
    private String isbn13;
    private int totalPage;
    private LocalDateTime publishedDate;
    private float avgRate;
    private String categoryName;
    private List<String> authors;
    private ReadState readState;
    private PageResponseDto<ReviewResponseDto> reviews;
}

File: backend/src/main/java/com/back/domain/book/book/dto/BookDetailDto.kt
package com.back.domain.book.book.dto

import com.back.domain.bookmarks.constant.ReadState
import com.back.domain.review.review.dto.ReviewResponseDto
import com.back.global.dto.PageResponseDto
import java.time.LocalDateTime

data class BookDetailDto(
    val id: Long = 0L,
    val title: String? = null,
    val imageUrl: String? = null,
    val publisher: String? = null,
    val isbn13: String? = null,
    val totalPage: Int = 0,
    val publishedDate: LocalDateTime? = null,
    val avgRate: Float = 0f,
    val categoryName: String? = null,
    val authors: List<String> = emptyList(),
    val readState: ReadState? = null,
    val reviews: PageResponseDto<ReviewResponseDto>? = null
)